pages/borrow/[symbol].tsx

```diff
@@ -58,16 +64,135 @@ const Borrow: NextPage<
   )
 
   const collateralAmount = useMemo(
-    () => parseUnits(collateralValue, collateral?.decimals ?? 18),
-    [collateralValue, collateral?.decimals],
+    () => parseUnits(collateralValue, collateral?.underlying.decimals ?? 18),
+    [collateralValue, collateral?.underlying.decimals],
+  )
+  const collateralBalance = useMemo(
+    () => (collateral ? balances[collateral.underlying.address] : 0n),
```

Comment:
`collateralBalance` can be undefined.

`collateral ? balances[collateral.underlying.address] ?? 0n : 0n`